rdf4j/src/main/java/org/apache/commons/rdf/rdf4j/RDF4JGraphLike.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.rdf.rdf4j;
+
+import java.util.Optional;
+
+import org.apache.commons.rdf.api.BlankNodeOrIRI;
+import org.apache.commons.rdf.api.GraphLike;
+import org.apache.commons.rdf.api.IRI;
+import org.apache.commons.rdf.api.RDFTerm;
+import org.apache.commons.rdf.api.TripleLike;
+import org.eclipse.rdf4j.model.Model;
+import org.eclipse.rdf4j.repository.Repository;
+
+/**
+ * Marker interface for RDF4J implementations of GraphLike.
+ * 
+ * @see RDF4JGraph
+ * 
+ */
+public interface RDF4JGraphLike<T extends TripleLike<BlankNodeOrIRI, IRI, RDFTerm>>
```

Comment:
(not sure where to comment)

"Generalised RDF" is actually a concept in RDF 1.1. The generalized RDF 1.1 came about because JSON-LD as bnodes for graph names ... and the perma-thread on literals-as-subjects.  So the WG put in the terminology for all of that.

RDF 1.1 (Generalized) does not have "quad", just RDF Dataset, Graphs, Triples and RDF Terms.  Quads are reasonably natural in that a dataset is a set of triples (default graph) and an set of quads (named graphs).

Commons RDF can either a common subset of all it integrates or try to espouse it's own take.  At the moment, it seems to me, getting it users matters.  Going too much into the details, and instead being almost a bit loose in the interests of time, could be better way to go.

As a usage : a SHACL Core engine would be a good target. It is an algorithm that can be used with various systems.  SHACL Core does not use SPARQL; some definitions in SHACL Core give both a textual and SPARQL version.